policy/modules/system/systemd.te

```diff
@@ -1137,6 +1158,42 @@ optional_policy(`
 	dbus_watch_system_bus_runtime_named_sockets(systemd_resolved_t)
 ')
 
+#########################################
+#
+# systemd selinux-autorelabel local policy
+#
+
+allow systemd_selinux_autorelabel_t self:fifo_file rw_fifo_file_perms;
+allow systemd_selinux_autorelabel_t self:process getsched;
+
+corecmd_exec_bin(systemd_selinux_autorelabel_t)
+corecmd_exec_shell(systemd_selinux_autorelabel_t)
+
+files_read_etc_files(systemd_selinux_autorelabel_t)
+files_purge_tmp(systemd_selinux_autorelabel_t)
+# for /.autorelabel
+files_read_etc_runtime_files(systemd_selinux_autorelabel_t)
+files_delete_boot_flag(systemd_selinux_autorelabel_t)
+
+kernel_read_system_state(systemd_selinux_autorelabel_t)
+
+term_use_console(systemd_selinux_autorelabel_t)
+
+init_use_fds(systemd_selinux_autorelabel_t)
+# for triggering reboot
+init_stream_connect(systemd_selinux_autorelabel_t)
+init_reboot_system(systemd_selinux_autorelabel_t)
+init_read_state(systemd_selinux_autorelabel_t)
+
+miscfiles_read_localization(systemd_selinux_autorelabel_t)
+
+seutil_read_config(systemd_selinux_autorelabel_t)
+seutil_domtrans_setfiles(systemd_selinux_autorelabel_t)
+# for genhomedircon
+seutil_domtrans_semanage(systemd_selinux_autorelabel_t)
+
+quota_domtrans(systemd_selinux_autorelabel_t)
```

Comment:
This should be optional unless it absolutely can't work without this.

Reply:
I'll have to test if there's a way to work around this, the script invokes `[ -x "/sbin/quotaoff" ] && /sbin/quotaoff -aug`.
Same with genhomdircon but it might be possible to replace it with read/write access to /etc/selinux.

Reply:
If it's only running quotaoff/on only if the tool is executable, I'd think this would be fine as optional. If someone has a policy with no quota module but quotas enable somehow, there will be more problems than this tool.

Reply:
Oh I misunderstood you.